masonry_core/src/app/render_root.rs

```diff
@@ -268,6 +269,57 @@ pub enum RenderRootSignal {
     ShowWindowMenu(LogicalPosition<f64>),
     /// The widget picker has selected this widget.
     WidgetSelectedInInspector(WidgetId),
+    /// A new layer should be created with the widget as root.
+    NewLayer(NewWidget<dyn Widget>, Point),
+    /// The layer with the given widget as root should be removed.
+    RemoveLayer(WidgetId),
+    /// The layer with the given widget as root should be repositioned to the specified point.
+    RepositionLayer(WidgetId, Point),
+}
+
+impl std::fmt::Debug for RenderRootSignal {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::Action(action, widget_id) => f
+                .debug_tuple("Action")
+                .field(action)
+                .field(widget_id)
+                .finish(),
+            Self::StartIme => write!(f, "StartIme"),
+            Self::EndIme => write!(f, "EndIme"),
+            Self::ImeMoved(origin, size) => {
+                f.debug_tuple("ImeMoved").field(origin).field(size).finish()
+            }
+            Self::ClipboardStore(item) => f.debug_tuple("ClipboardStore").field(item).finish(),
+            Self::RequestRedraw => write!(f, "RequestRedraw"),
+            Self::RequestAnimFrame => write!(f, "RequestAnimFrame"),
+            Self::TakeFocus => write!(f, "TakeFocus"),
+            Self::SetCursor(icon) => f.debug_tuple("SetCursor").field(icon).finish(),
+            Self::SetSize(size) => f.debug_tuple("SetSize").field(size).finish(),
+            Self::SetTitle(title) => f.debug_tuple("SetTitle").field(title).finish(),
+            Self::DragWindow => write!(f, "DragWindow"),
+            Self::DragResizeWindow(d) => f.debug_tuple("DragResizeWindow").field(d).finish(),
+            Self::ToggleMaximized => write!(f, "ToggleMaximized"),
+            Self::Minimize => write!(f, "Minimize"),
+            Self::Exit => write!(f, "Exit"),
+            Self::ShowWindowMenu(pos) => f.debug_tuple("ShowWindowMenu").field(pos).finish(),
+            Self::WidgetSelectedInInspector(widget_id) => f
+                .debug_tuple("WidgetSelectedInInspector")
+                .field(widget_id)
+                .finish(),
+            Self::NewLayer(root_widget, pos) => f
+                .debug_tuple("NewLayer")
+                .field(&root_widget.widget.short_type_name())
+                .field(&pos)
+                .finish(),
```

Comment:
(Non-blocking) Rather than remove the Debug impl for RenderRootSignal, I'd rather add a Debug impl to NewWidget.